server/api.go

```diff
@@ -853,3 +851,182 @@ func (p *Plugin) updateSettings(w http.ResponseWriter, r *http.Request, userID s
 	resp, _ := json.Marshal(info.Settings)
 	w.Write(resp)
 }
+
+func (p *Plugin) getRepositories(w http.ResponseWriter, r *http.Request, userID string) {
+	info, err := p.getGitHubUserInfo(userID)
+	if err != nil {
+		writeAPIError(w, err)
+		return
+	}
+
+	githubClient := p.githubConnect(*info.Token)
+
+	ctx := context.Background()
+	org := p.getConfiguration().GitHubOrg
+
+	var allRepos []*github.Repository
+	opt := github.ListOptions{PerPage: 50}
+
+	if org == "" {
+		for {
+			repos, resp, err := githubClient.Repositories.List(ctx, "", &github.RepositoryListOptions{ListOptions: opt})
+			if err != nil {
+				mlog.Error(err.Error())
+				return
+			}
+			allRepos = append(allRepos, repos...)
+			if resp.NextPage == 0 {
+				break
+			}
+			opt.Page = resp.NextPage
+		}
+	} else {
+		for {
+			repos, resp, err := githubClient.Repositories.ListByOrg(ctx, org, &github.RepositoryListByOrgOptions{Sort: "full_name", ListOptions: opt})
+			if err != nil {
+				mlog.Error(err.Error())
+				return
+			}
+			allRepos = append(allRepos, repos...)
+			if resp.NextPage == 0 {
+				break
+			}
+			opt.Page = resp.NextPage
+		}
+	}
+
+	// Only send down fields to client that are needed
+	type RepositoryResponse struct {
+		Name     string `json:"name,omitempty"`
+		FullName string `json:"full_name,omitempty"`
+	}
+
+	response := make([]RepositoryResponse, len(allRepos))
+	for i, r := range allRepos {
+		response[i].Name = r.GetName()
+		response[i].FullName = r.GetFullName()
+	}
+
+	resp, _ := json.Marshal(response)
+	w.Write(resp)
+}
+
+func (p *Plugin) createIssue(w http.ResponseWriter, r *http.Request, userID string) {
+	type IssueRequest struct {
+		Title  string `json:"title"`
+		Body   string `json:"body"`
+		Repo   string `json:"repo"`
+		PostId string `json:"post_id"`
+	}
+
+	// get data for the issue from the request body and fill IssueRequest object
+	issue := &IssueRequest{}
+	if err := json.NewDecoder(r.Body).Decode(&issue); err != nil {
```

Comment:
Can we add this italics text at the end of the issue description:

> _Created in Mattermost from [this post](somewhere.com)_

There is similar functionality in the Jira plugin's feature

Reply:
That is already the case

![Screenshot from 2020-05-09 01-29-04](https://user-images.githubusercontent.com/16541325/81458345-35185680-919a-11ea-8d07-87140144c7db.png)